### DISCUSSION TITLE
Curious about how you compiled ngspice to WASM

### DISCUSSION THREAD

**Discussion by michaelbzhu** [danchitnis/EEsim#6] [Q&A] [answered]
Hey really cool project! 

I'm thinking of building a similar project but a much simpler version that only uses RLC components (I've only taken the intro EE course for my undergrad so I don't know too much beyond that). 

I've been looking into compiling [SPICE](http://bwrcs.eecs.berkeley.edu/Classes/IcBook/SPICE/) and/or [ngspice](https://sourceforge.net/p/ngspice/ngspice/ci/master/tree/) into WASM so I'm curious how you implemented that part.

Are there any resources that you found useful that you can point me to?

**Comment by danchitnis** [comment, marked as answer]
For now, as of today, you can compile ngspice codebase to WASM without any hacking. See [here](https://github.com/danchitnis/ngspice) for the build tools and example. There is a discussion thread on the previous hacks which you can find [here](https://github.com/danchitnis/ngspice#details). If you want only RLC, then you can remove the extra models from the ngspice so you have a smaller (and perhaps) faster WASM. 

Notice that the original [SPICE3f5](https://ptolemy.berkeley.edu/projects/embedded/pubs/downloads/spice/) from Berkeley is slightly different from ngspice as it is missing the contemporary models. If you want only RLC as a practice then the original SPICE3f5 might be more suitable.
